frontend/src/App.tsx

```diff
@@ -1,58 +1,41 @@
-import { useCallback, useEffect, useMemo, useState } from "react";
+import { useCallback, useMemo, useState } from "react";
 import { InformationCircleIcon } from "@heroicons/react/24/outline";
 import { Chat } from "./components/Chat";
 import { ChatList } from "./components/ChatList";
 import { Layout } from "./components/Layout";
 import { NewChat } from "./components/NewChat";
-import { Chat as ChatType, useChatList } from "./hooks/useChatList";
+import { useChatList } from "./hooks/useChatList";
 import { useSchemas } from "./hooks/useSchemas";
 import { useStreamState } from "./hooks/useStreamState";
-import { useConfigList } from "./hooks/useConfigList";
+import {
+  useConfigList,
+  Config as ConfigInterface,
+} from "./hooks/useConfigList";
 import { Config } from "./components/Config";
 import { MessageWithFiles } from "./utils/formTypes.ts";
-import { TYPE_NAME } from "./constants.ts";
+import { Route, Routes, useNavigate } from "react-router-dom";
+import { NotFound } from "./components/NotFound.tsx";
 
 function App() {
+  const navigate = useNavigate();
   const [sidebarOpen, setSidebarOpen] = useState(false);
-  const { configSchema, configDefaults } = useSchemas();
-  const { chats, currentChat, createChat, enterChat } = useChatList();
-  const { configs, currentConfig, saveConfig, enterConfig } = useConfigList();
+  const { chats, createChat } = useChatList();
+  const { configs, saveConfig } = useConfigList();
   const { startStream, stopStream, stream } = useStreamState();
-  const [isDocumentRetrievalActive, setIsDocumentRetrievalActive] =
-    useState(false);
+  const { configSchema, configDefaults } = useSchemas();
 
-  useEffect(() => {
-    let configurable = null;
-    if (currentConfig) {
-      configurable = currentConfig?.config?.configurable;
-    }
-    if (currentChat && configs) {
-      const conf = configs.find(
-        (c) => c.assistant_id === currentChat.assistant_id,
-      );
-      configurable = conf?.config?.configurable;
-    }
-    const agent_type = configurable?.["type"] as TYPE_NAME | null;
-    if (agent_type === null || agent_type === "chatbot") {
-      setIsDocumentRetrievalActive(false);
-      return;
-    }
-    if (agent_type === "chat_retrieval") {
-      setIsDocumentRetrievalActive(true);
-      return;
-    }
-    const tools =
-      (configurable?.["type==agent/tools"] as { name: string }[]) ?? [];
-    setIsDocumentRetrievalActive(tools.some((t) => t.name === "Retrieval"));
-  }, [currentConfig, currentChat, configs]);
+  const [currentChatId, setCurrentChatId] = useState<string | null>(null);
```

Comment:
why not store this in url query params?

Reply:
I messed up a bit with the router. `useParams` should give this data but the router needs to be raised from App.tsx to get the router `useParams` work.
The PR is becoming so heavy that I decided to make a working version now with the routes and url based consistent state. Good enough for now?
Next week I can get back to making it better?